api/src/services/applications/create_application.py

```diff
@@ -67,8 +69,20 @@ def create_application(db_session: db.Session, competition_id: UUID, user: User)
                 ],
             )
 
+    # Get default application name if not provided
+    if (
+        application_name is None
+        and competition.opportunity
+        and competition.opportunity.opportunity_number
+    ):
+        application_name = competition.opportunity.opportunity_number
+
```

Comment:
I think we can just simplify to this?
```suggestion
    # Get default application name if not provided
    if application_name is None:
        application_name = competition.opportunity.opportunity_number
```
Opportunity can't be none, opportunity number shouldn't ever be none, but even if it is, grabbing None instead of using the "None" in application name is the same thing.